grails-app/domain/org/pih/warehouse/core/Location.groovy

```diff
@@ -285,6 +285,24 @@ class Location implements Comparable<Location>, java.io.Serializable {
         Location.list().findAll { it.isDepotWardOrPharmacy() }.sort { it.name }
     }
 
+    Location getDefaultLocation() {
+        return null
+    }
+
+    /**
+     * Find any internal location with the given name.
+     * @param name
+     * @return
+     */
+    Location getInternalLocation(String name) {
+        if (name?.equalsIgnoreCase(Constants.DEFAULT_BIN_LOCATION_NAME)) {
+            return defaultLocation
+        }
+
+        // FIXME If location.name is not unique then we need to do some error handling
+        return locations.find { it.name.equalsIgnoreCase(name) }
```

Comment:
if this is an internal location, you don't need to do an `isInternalLocation()` check?